dev/bots/allowlist.dart

```diff
@@ -33,6 +33,7 @@ const Set<String> kCorePackageAllowList = <String>{
   'fuchsia_remote_debug_protocol',
   'integration_test',
   'intl',
+  'js',
```

Comment:
Have you cleared this according to the comment on top? Should the "clearer" be a reviewer in this PR too?

```
/// Before adding a new Dart Team owned dependency to this set, please clear with natebosch@
/// or jakemac53@. For other packages please contact hixie@ or zanderso@.
```

Reply:
Whoops, Good catch. I actually did talk to @jakemac53 offline, but we need the official stamp.

@jakemac53 could you take a look for Dart team?

Reply:
LGTM